File: tests/TypedUsing.cs
using System;

namespace Demo {
    public partial class TypedUsing {
        public static IDisposable GetRes() {
            IDisposable result;
            result = null;
            return result;
        }
        public static void DoStuff() {
            using (IDisposable res = GetRes()) Console.WriteLine(res);
        }
    }
}

File: tests/TypedUsing.pas
namespace Demo;

type
  TypedUsing = class
  public
    class method GetRes: IDisposable;
    class method DoStuff;
  end;

implementation
uses System;

class method TypedUsing.GetRes: IDisposable;
begin
  result := nil;
end;

class method TypedUsing.DoStuff;
begin
  using res: IDisposable := GetRes() do
    Console.WriteLine(res);
end;

end.
